### PR TITLE
Add priority

### PR DESCRIPTION
**Pull request by Melvin-Tan**
#93 
my first UI issue yay

### CODE REVIEW

File: src/main/java/seedu/toluist/controller/UpdateTaskController.java [hunk 1]
```diff
@@ -57,10 +59,17 @@ public CommandResult execute(String command) {
 
         boolean isFloating = tokens.containsKey(TaskTokenizer.TASK_FLOATING_KEYWORD);
 
+        Optional<Boolean> isHighPriority = Optional.empty();
+        if (tokens.containsKey(TaskTokenizer.TASK_PRIORITY_KEYWORD)) {
+            String taskPriority = tokens.get(TaskTokenizer.TASK_PRIORITY_KEYWORD);
+            isHighPriority = Optional.of(taskPriority.equals("high"));
```
Comment (louislai) [inline src/main/java/seedu/toluist/controller/UpdateTaskController.java]: so doing `priority/very high` will make the priority low?

Comment (Melvin-Tan) [reply src/main/java/seedu/toluist/controller/UpdateTaskController.java]: so u want something that checks for contain "high"?
cuz i thought we just need to write in user doc that we only accept "high" or "low"

OH YA MAGIC CONSTANT LEMME REFACTOR LATER

Comment (louislai) [reply src/main/java/seedu/toluist/controller/UpdateTaskController.java]: `cuz i thought we just need to write in user doc that we only accept "high" or "low"` then it's an error condition that has no explicit error messages

Comment (louislai) [reply src/main/java/seedu/toluist/controller/UpdateTaskController.java]: containing `high` is noa good solution. `not very high`

File: src/main/java/seedu/toluist/model/Task.java [hunk 2]
```diff
@@ -237,6 +243,16 @@ public void setTaskPriority(TaskPriority priority) {
         this.priority = priority;
     }
 
+    public void setTaskPriority(String priorityString) {
+        if (priorityString.equalsIgnoreCase(HIGH_PRIORITY_STRING)) {
+            this.priority = TaskPriority.HIGH;
```
Comment (louislai) [inline src/main/java/seedu/toluist/model/Task.java]: can use the other setter here?

Comment (Melvin-Tan) [reply src/main/java/seedu/toluist/model/Task.java]: done

File: src/test/java/guitests/UpdateTaskCommandTest.java [hunk 3]
```diff
@@ -129,7 +135,7 @@ public void updateEvents() {
         LocalDateTime newerStartDate = DateTimeUtil.parseDateString("15 Mar 2017, 12pm");
         LocalDateTime newerEndDate = DateTimeUtil.parseDateString("15 Mar 2017, 1pm");
         command = "update " + eventIndex + " " + newerTaskDescription +
-                  " to/" + newerEndDate + " tags/tag1 "  + " from/" + newerStartDate;
+                  " priority/low to/" + newerEndDate + " tags/tag1 "  + " from/" + newerStartDate;
```
Comment (louislai) [inline src/test/java/guitests/UpdateTaskCommandTest.java:138]: add a test for `priority/ low   high` and another one for `priority/`

**Comment by louislai** [comment]
should make low/high case insensitive as well (in fact the tokenizer should allows case insensitive keywords as well....)

**Comment by Melvin-Tan** [comment]
if (priorityString.equalsIgnoreCase(HIGH_PRIORITY_STRING))

already did case insensitive o.o

**Comment by louislai** [comment]
oh wait `priority/HIGH` works. How did that happen

File: src/test/java/guitests/UpdateTaskCommandTest.java (hunk 3 above)
Comment (Melvin-Tan) [reply src/test/java/guitests/UpdateTaskCommandTest.java:138]: done, in both update and add

File: src/test/java/guitests/UpdateTaskCommandTest.java [hunk 4]
```diff
@@ -307,4 +313,23 @@ public void updateTaskAfterMarkingComplete() {
...
+        String taskDescription = "attend CS2103T tutorial";
+        String command = "add " + taskDescription;
+        commandBox.runCommand(command);
+        Task task = new Task(taskDescription);
+        assertTrue(isTaskShown(task));
+
+        command = "update " + eventIndex + " priority/high low";
+        commandBox.runCommand(command);
+        assertTrue(isTaskShown(task));
+
```
Comment (louislai) [inline src/test/java/guitests/UpdateTaskCommandTest.java:330]: check that the result messages are correct as well. `assertResultMessage`

**Pull request closed without merging**